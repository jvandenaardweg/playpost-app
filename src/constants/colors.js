const tintColor = '#037DE2';
const black = '#000000';
const grayDark = '#333C46';
const grayLight = '#cccccc';
const grayLightest = '#EEF0F4';
const appBackground = '#EBEBEB';
const white = '#ffffff';
const blue = tintColor;
const red = 'red';
const orange = '#EAEB5E';
const orangeDark = '#666804';

export default {
  tintColor,
  black,
  grayDark,
  grayLight,
  grayLightest,
  white,
  blue,
  red,
  orange,
  appBackground,

  tabIconDefault: grayLight,
  tabIconSelected: tintColor,
  tabBar: white,

  articleBackground: white,

  titleDefault: black,
  paragraphDefault: black,
  paragraphGrayed: grayDark,
  borderDefault: grayLightest,

  controlButtonDefault: black,
  controlButtonActive: tintColor,

  errorBackground: red,
  errorText: white,
  warningBackground: orange,
  warningText: orangeDark,
  noticeBackground: tintColor,
  noticeText: white,
};
