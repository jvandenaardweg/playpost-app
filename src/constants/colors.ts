const tintColor = '#037DE2';
const black = '#000000';
const grayDarkest = '#222222';
const grayDarker = '#444444';
const grayDark = '#888888';
const gray = '#bfbfbf';
const grayLight = '#e5e5e5';
const grayLightest = '#EEF0F4';
const appBackground = '#F1F1F1';
const white = '#ffffff';
const blue = tintColor;
const red = '#FF1017';
const redDark = '#b52424';
const orange = '#EAEB5E';
const orangeDark = '#666804';
const green = '#03A87C';

export default {
  tintColor,
  black,
  grayDarkest,
  grayDarker,
  grayDark,
  gray,
  grayLight,
  grayLightest,
  white,
  blue,
  red,
  redDark,
  orange,
  green,
  appBackground,

  tabIconDefault: gray,
  tabIconSelected: black,
  tabBar: white,

  articleBackground: white,

  titleDefault: black,
  paragraphDefault: grayDarkest,
  paragraphGrayed: grayDark,
  borderDefault: grayLight,

  controlButtonDefault: black,
  controlButtonActive: tintColor,

  errorBackground: red,
  errorText: white,
  warningBackground: orange,
  warningText: orangeDark,
  noticeBackground: tintColor,
  noticeText: white,
};
