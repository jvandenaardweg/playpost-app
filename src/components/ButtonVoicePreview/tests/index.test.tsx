import React from 'react';
import { fireEvent, render, RenderAPI } from 'react-native-testing-library';

import colors from '../../../constants/colors';
import { ButtonVoicePreview } from '../index';

const onPressHandler = jest.fn();

const defaultProps = {
  isLoading: false,
  isPlaying: false,
  isActive: false,
  isAvailable: false,
  onPress: onPressHandler
}

describe('ButtonVoicePreview', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      const props = {
        ...defaultProps
      }

      wrapper = render(<ButtonVoicePreview {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should fire onPress when the button is pressed', () => {
      fireEvent.press(wrapper.getByTestId('ButtonViewPreview-TouchableOpacity'));
      expect(onPressHandler).toHaveBeenCalledTimes(1);
    });

    it('should render a ActivityIndicator when loading', () => {
      const props = {
        ...defaultProps,
        isLoading: true
      }

      wrapper.update(<ButtonVoicePreview {...props} />);
      expect(wrapper.getByTestId('ButtonViewPreview-ActivityIndicator')).toBeTruthy();
    });

    it('should disable the button when loading', () => {
      const props = {
        ...defaultProps,
        isLoading: true
      }

      wrapper.update(<ButtonVoicePreview {...props} />);
      expect(wrapper.getByTestId('ButtonViewPreview-TouchableOpacity').props.disabled).toBe(true);
    });

    it('should render a different button color when playing', () => {
      const props = {
        ...defaultProps,
        isPlaying: true
      }

      wrapper.update(<ButtonVoicePreview {...props} />);
      expect(wrapper.getByTestId('ButtonViewPreview-TouchableOpacity').props.style[1].backgroundColor).toBe(colors.tintColor);
    });

    it('should render a different button color when a voice preview is available', () => {
      const props = {
        ...defaultProps,
        isActive: false,
        isAvailable: true
      }

      wrapper.update(<ButtonVoicePreview {...props} />);
      expect(wrapper.getByTestId('ButtonViewPreview-TouchableOpacity').props.style[2].backgroundColor).toBe(colors.black);
    });

    it('should render a different button color when active', () => {
      const props = {
        ...defaultProps,
        isActive: true
      }

      wrapper.update(<ButtonVoicePreview {...props} />);
      expect(wrapper.getByTestId('ButtonViewPreview-TouchableOpacity').props.style[1].backgroundColor).toBe(colors.tintColor);
    });

    it('should render a different button color when no voice preview is available', () => {
      const props = {
        ...defaultProps,
        isAvailable: false
      }

      wrapper.update(<ButtonVoicePreview {...props} />);
      expect(wrapper.getByTestId('ButtonViewPreview-TouchableOpacity').props.style[0].backgroundColor).toBe(colors.gray);
    });

    it('should render a pause Icon when playing', () => {
      const props = {
        ...defaultProps,
        isPlaying: true
      }

      wrapper.update(<ButtonVoicePreview {...props} />);
      expect(wrapper.getByTestId('ButtonViewPreview-Icon-pause')).toBeTruthy();
    });

    it('should render a pause Icon when playing', () => {
      const props = {
        ...defaultProps,
        isPlaying: true
      }

      wrapper.update(<ButtonVoicePreview {...props} />);
      expect(wrapper.getByTestId('ButtonViewPreview-Icon-pause')).toBeTruthy();
    });
  });
});
