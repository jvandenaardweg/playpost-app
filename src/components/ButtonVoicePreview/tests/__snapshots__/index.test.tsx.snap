// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonVoicePreview rendering should render correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#C5C5C5",
      "borderRadius": 32,
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "width": 32,
    }
  }
  testID="ButtonViewPreview-TouchableOpacity"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 9,
        },
        Object {
          "color": "#ffffff",
          "position": "relative",
          "right": -1,
        },
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "900",
        },
      ]
    }
    testID="ButtonViewPreview-Icon-play"
  >
    
  </Text>
</View>
`;
